src/main/java/gameframework/motion/overlapping/OverlapProcessorDefaultImpl.java

```diff
@@ -104,17 +91,30 @@ protected void computeOneOverlap(Overlappable movableOverlappable,
 										.getDirection(),
 								-((GameMovable) targetOverlappable)
 										.getSpeedVector().getSpeed()));
-				boundingBoxTarget = targetShape.getBounds();
-
-				if (boundingBoxOverlappable.intersects(boundingBoxTarget)) {
-					targetArea = new Area(targetShape);
-					targetArea.intersect(overlappableArea);
-					if (!targetArea.isEmpty()) {
-						// NOTE I don't see how this test could fail
-						overlaps.add(new Overlap(movableOverlappable,
-								targetOverlappable));
-					}
-				}
+				overlapRectangle(movableOverlappable, overlaps,
+						targetOverlappable,
+						targetShape);
+			}
+		}
+	}
+
+	private void overlapRectangle(Overlappable movableOverlappable, Vector<Overlap> overlaps, 
```

Comment:
please use `protected` instead of `private` because subclasses might want to adapt it.